lib/babe/babe.go

```diff
@@ -185,13 +179,16 @@ func (b *Service) setEpochData(cfg *ServiceConfig, genCfg *types.BabeConfigurati
 		}
 	}
 
-	if cfg.Threshold == nil {
+	if cfg.ThresholdDenominator == 0 {
 		b.epochData.threshold, err = CalculateThreshold(genCfg.C1, genCfg.C2, len(b.epochData.authorities))
 		if err != nil {
 			return err
 		}
 	} else {
-		b.epochData.threshold = cfg.Threshold
+		b.epochData.threshold, err = CalculateThreshold(cfg.ThresholdNumerator, cfg.ThresholdDenominator, len(b.epochData.authorities))
+		if err != nil {
+			return err
+		}
```

Comment:
nitpick, but you could move the err check below the if/else

Reply:
Ahh yeah should of spotted that, thanks!